tools/blobcp.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+    "flag"
+    "fmt"
+    "io"
+    "log"
+    "os"
+    "path/filepath"
+    "sync"
+    "sync/atomic"
+    "time"
+)
+
+var workers = flag.Int("n", 32, "Number of workers")
+var source = flag.String("s", "", "Source path to walk through")
+var dest = flag.String("d", "", "Destination to copy to")
+var include = flag.String("pattern", "", "Path pattern to match in source path")
+var processed int64 = 0
+
+func main() {
+    flag.Parse()
+    if *source == "" {
+        log.Printf("-s source path is missing")
+        flag.PrintDefaults()
+    } else if *dest == "" {
+        log.Printf("-d destination path is missing")
+        flag.PrintDefaults()
+    }
+
+    pending := sync.WaitGroup{}
+    files := make(chan string)
+
+    go func() {
```

Comment:
do we need it?

Reply:
What do you mean ? Doesn't this have to be in a go routine ?